pkg/gds/gds.go

```diff
@@ -470,16 +467,31 @@ func (s *GDS) VerifyContact(ctx context.Context, in *api.VerifyContactRequest) (
 		return nil, status.Error(codes.NotFound, "could not find associated VASP record by ID")
 	}
 
-	// Search through the contacts to determine the contacts verified by the supplied token.
-	prevVerified := 0
+	// Retrieve contactEmail address from one of the supplied contacts.
+	var contactEmail string
+	if contactEmail, _ = GetContactInfo(vasp); contactEmail == "" {
+		log.Error().Err(errors.New("no contact email address found")).Msg("incorrect access on validated VASP")
+		return nil, status.Error(codes.InvalidArgument, "no email address in supplied VASP contacts")
+	}
+
+	var contact *models.Contact
+	if contact, err = s.db.RetrieveContact(ctx, contactEmail); err != nil {
+		log.Warn().Err(err).Str("contact", contactEmail).Msg("could not retrieve contact")
+		return nil, status.Error(codes.NotFound, "could not find associated contact record by email")
+	}
+
 	found := false
-	contactEmail := ""
+	prevVerified := 0
 
+	// Search through the contacts to determine the contacts verified by the supplied token.
```

Comment:
Ok I think I'm getting my wires crossed here. We are storing contact verifications on the new contact model now right? Do we need to be fetching those from the database rather than the contacts on the VASP?